### ISSUE TITLE
ProtocolHandler API: handler is not called in extension development setup

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#48511]
testing #48427 on macOS:

- create hello world extension as 'protocol-handler'
- make it activate on `'*'`
- in the activate function register a uri handler like this:
  `context.subscriptions.push(vscode.window.registerProtocolHandler(new myUriHandler()));`
- the handler is this: 
  ```
  class myUriHandler implements vscode.ProtocolHandler {
    handleUri(uri: vscode.Uri) {
        vscode.window.showErrorMessage(`uri: ${uri.toString()}`, { modal: true });
    }
  } 
  ```
- press F5 to launch extension
- In an external terminal I run this:
  `open vscode-insiders://andreweinand.protocol-handler/foo`

In VS Code's window for the 'protocol-handler' project (not the EH window) the following alerts shows up:
![2018-04-24_12-06-38](https://user-images.githubusercontent.com/1898161/39180856-47687938-47b8-11e8-8c01-4cbaa38172dd.png)

After pressing 'Yes' nothing happens.

I suspect that the wrong instance of VS Code receives the uri. 

When installing the extension in VS Code, the handler is called as expected.

**Comment by joaomoreno** [comment]
This seems to be going to the focused window only. We should make it go to all windows. But something else is wrong here, because the extension doesn't get the URI even when using the `Developer: Open URL` command. It is possible that the extension has a different ID in our services because it is the currently debugged extension.

**Comment by sbatten** [comment]
fyi for Windows, both scenarios (from external terminal and command palette), work as expected

Version 1.23.0-insider
Commit b845ce918f2689344795e845ae6ef691916c3079

**Comment by joaomoreno** [comment]
This has been fixed along with #48608 and #48609
